app/helpers/sessions_helper.rb

```diff
@@ -1,17 +1,23 @@
 module SessionsHelper
+  
   # Returns the current logged-in user (if any).
   def current_user
     if (user_id = session[:user_id])
       user ||= User.find_by(id: user_id)
         @current_user||= user if (user.session_token!=session['session_token'])
     elsif (user_id = cookies.encrypted[:user_id])
       user = User.find_by(id: user_id)
-      if user && user.authenticated?(cookies[:remember_token])
+      if user && user.authenticated?(:remember, cookies[:remember_token])
```

Comment:
when you start chaining conditions, see if you can extract the lot into a well named method :)